src/crypto/NoteEncryption.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+See the Zcash protocol specification for more information.
+https://github.com/zcash/zips/blob/master/protocol/protocol.pdf
+
+Zcash allows users to send and receive value using notes. In order
+to protect the privacy of the parties involved, this note is not
+public, and for verification purposes, is cryptographically
+committed. This commitment is public, and lives in the transaction.
+
+However, recipients must be able to obtain values of the note,
+including the value `v`, a secret `rho`, and the trapdoor `r`
+otherwise they will be unable to spend it. In addition, we would
+like to send a 64-byte memo to the recipient in the transaction.
+
+In order to do this privately, we require the use of an IND-CCA2
+secure encryption scheme. However, off-the-shelf schemes were
```

Comment:
We also require key-privacy (ask @daira what the formal name of that property is).